wormhole/module.py

```diff
@@ -49,147 +47,152 @@ def __init__(self, bot: Strawberry):
+                f"Cannot set slowmode in {', '.join(forbidden)}",
             )
             if itx:
                 await itx.response.send_message(
                     _(
                         itx,
-                        "I do not have proper permissions to set slow mode. Some channel may need manual intervention.",
+                        "Lacking permissions to set slowmode in some channels.",
                     ),
                     ephemeral=True,
                 )
         else:
             await bot_log.info(
                 itx.user if itx else None,
                 itx.channel if itx else None,
-                f"Wormhole slow mode set to {delay} seconds.",
+                f"Slowmode set to {delay}s.",
             )
             if itx:
                 await itx.response.send_message(
-                    _(itx, "Slow mode set to {delay} seconds.").format(delay=delay),
+                    _(itx, "Slowmode set to {delay}s.").format(delay=delay),
                     ephemeral=True,
                 )
 
-    # LISTENERS
-
+    # LISTENER
     @commands.Cog.listener()
     async def on_message(self, message: discord.Message):
-        """Main message relay logics."""
-        # Ignore bot messages
-        if message.author.bot:
-            return
-
-        # Ignore commands
-        if message.content.startswith(self.bot.command_prefix):
+        """Relay text and images to configured wormhole channels."""
+        if message.author.bot or message.content.startswith(self.bot.command_prefix):
             return
-
-        # Only proceed if this channel is registered as a wormhole
         if message.channel.id not in self.wormhole_channels:
             return
 
+        # Delete original message
         try:
-            await message.delete()  # Delete original user message
+            await message.delete()
         except discord.Forbidden:
             await bot_log.warning(
-                message.author,
-                message.channel,
-                "Missing permissions to delete message.",
+                message.author, message.channel, "Cannot delete message."
             )
 
-        formatted_message = await self._message_formatter(message)  # Format message
-
-        # Send to all wormhole channels
-        for channel in self.wormhole_channels:
-            target_channel = self.bot.get_channel(channel)
-            if target_channel:
-                try:
-                    await target_channel.send(formatted_message)
-                except discord.Forbidden:
-                    await bot_log.warning(
-                        message.author,
-                        target_channel,
-                        "Missing permissions to send the message.",
-                    )
-
-    # COMMANDS
+        formatted, thumbnail = await self._message_formatter(message)
```

Comment:
That thumbnail function adds unnecessary complexity. I'd like to keep the sending function as small and compact as possible for performance reasons.